### PR TITLE
doc/TPC-1320-split-mobile-sdk-doc-by-platform

### PR DESCRIPTION
**Pull request by Judas**
#### Description
This PR splits the Mobile SDK documentation into 4 sections, one per mobile platform (Android, iOS, ReactNative, Flutter)

### CODE REVIEW

**Review by EtienneRd — COMMENTED** [review]
The satisfaction survey gif does not show the NPS fix

File: docs/mobile-sdk.md [hunk 1]
```diff
@@ -23,137 +25,52 @@ The iAdvize operator desk is the place where the conversations that are assigned
...
+
 allprojects {
   repositories {
     maven { url = uri("https://raw.github.com/iadvize/iadvize-android-sdk/master") }
   }
 }
 </pre>
 
-Add the iAdvize Messenger SDK dependency inside your module-level Gradle build file:
+Add the iAdvize Messenger SDK dependency inside your module-level Gradle build file (replace `x.y.z` by the SDK latest version available):
```
Comment (alexduf) [inline docs/mobile-sdk.md:63]:
```suggestion
Add the iAdvize Messenger SDK dependency inside your module-level Gradle build file (replace `x.y.z` by the latest SDK version available):
```

File: docs/mobile-sdk.md [hunk 2]
```diff
@@ -163,41 +80,22 @@ dependencies {
...
-    }
-}
-</pre>
- 
-⌨️ **In-context example:** [SDK Activation](https://github.com/iadvize/iadvize-ios-sdk/blob/master/Example/IAdvizeSwiftExample/IAdvizeSwiftExample/Source/AppDelegate%2BiAdvize.swift#L61)
-
-##### Android
-
-On Android there is an additional step required before activating the SDK. You will need to provide a reference to your application object and initiate the SDK with it.
+Before activating the SDK, you will need to provide a reference to your application object and initiate the SDK with it.
```
Comment (alexduf) [inline docs/mobile-sdk.md:93]:
```suggestion
Before activating the SDK, you will need to provide a reference to your application object and initialize the SDK with it.
```

File: docs/mobile-sdk.md [hunk 3]
```diff
@@ -215,9 +113,7 @@ class App : Application() {
 }
 </pre>
 
-> *⚠️ If you are developing an application using an hybrid framework based on a webview, you will need to do this call to `initiate` before the main webview activity is created, that means in the native Android app code, and not from the hybrid framework plugin. As the SDK relies on observing activities lifecycles to show/hide the chat button, not doing so will result in chat button not showing when it should.*
-
-After that you can activate the SDK:
+After that you can activate the SDK using the `activate` function with your `projectId` (see the [Prerequisites](#⚙️-prerequisites) section above to get that identifier). You have access to callbacks in order to know if the SDK has been successfully activated. In case of an SDK activation failure the callback will give you the reason of the failure and you may want to retry later:
```
Comment (alexduf) [inline docs/mobile-sdk.md:116]:
```suggestion
Afterwards you can activate the SDK using the `activate` function with your `projectId` (see the [Prerequisites](#⚙️-prerequisites) section above to get that identifier). You have access to callbacks in order to know if the SDK has been successfully activated. In the case of an SDK activation failure the callback will give you the reason of the failure and you may want to retry later:
```

File: docs/mobile-sdk.md [hunk 4]
```diff
@@ -237,267 +133,175 @@ IAdvizeSDK.activate(
...
 
-> *⚠️ After you activate a rule and it succeeds (by displaying the button), those conditions are checked every 30 seconds to verify that the button should still be displayed or not. At the first failure from this periodic check, the button is hidden and the SDK stops verifying the conditions. It means that if the rule cannot be triggered (after the first call, or after any successive check), you will have to call the `activateTargetingRule` method again to restart the engagement process.*
-
-This schema describes the process when you activate a targeting rule on the iAdvize Messenger SDK:
+> *⚠️ After you activate a rule and it succeeds (by displaying the button), those conditions are checked every 30 seconds to verify that the button should still be displayed or not. At the first failure from this periodic check, the button is hidden and the SDK stops verifying the conditions. It means that if the rule cannot be triggered (after the first call, or after any successive check), you will have to call the `activateTargetingRule` (or `registerUserNavigation`) method again to restart the engagement process.*
```
Comment (alexduf) [inline docs/mobile-sdk.md:228]:
```suggestion
> *⚠️ After you activate a rule and it succeeds (by displaying the button), those conditions are checked every 30 seconds to verify that the button should still be displayed or not. Upon first failure from this periodic check, the button is hidden and the SDK stops verifying the conditions. It means that if the rule cannot be triggered (after the first call, or after any successive check), you will have to call the `activateTargetingRule` (or `registerUserNavigation`) method again in order to restart the engagement process.*
```

File: docs/mobile-sdk.md [hunk 5]
```diff
@@ -1044,4 +625,1744 @@ ReflectionHelpers.setStaticField(android.os.Build::class.java, "MODEL", "whateve
 ReflectionHelpers.setStaticField(android.os.Build::class.java, "MANUFACTURER", "whatever")
 </pre>
 
-Please also be sure to initiate the SDK during the unit tests setup  (see the [Setting up the SDK](#⚙️-setting-up-the-sdk) section above).
+Please also be sure to initiate the SDK during the unit tests setup (see the [Setting up the SDK](#⚙️-setting-up-the-sdk) section above).
```
Comment (alexduf) [inline docs/mobile-sdk.md:628]:
```suggestion
Please also be sure to initialize the SDK during the unit tests setup (see the [Setting up the SDK](#⚙️-setting-up-the-sdk) section above).
```

**Review by alexduf — CHANGES_REQUESTED** [review]

**Review by alexduf — APPROVED** [review]

**Pull request merged**